packages/sheets-ui/src/views/clipboard/ClipboardPopupMenu.tsx

```diff
@@ -0,0 +1,221 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ICommandInfo } from '@univerjs/core';
+import type { IPasteHookKeyType, IPasteOptionCache } from '../../services/clipboard/type';
+import { ICommandService, IUniverInstanceService, LocaleService, toDisposable, useDependency, useObservable } from '@univerjs/core';
+import { DropdownOverlay, DropdownProvider, DropdownTrigger } from '@univerjs/design';
+import { IRenderManagerService } from '@univerjs/engine-render';
+import { CheckMarkSingle, MoreDownSingle, PasteSpecial } from '@univerjs/icons';
+import clsx from 'clsx';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { SheetOptionalPasteCommand } from '../../commands/commands/clipboard.command';
+import { SetScrollOperation } from '../../commands/operations/scroll.operation';
+import { useActiveWorkbook } from '../../components/hook';
+import { getSheetObject } from '../../controllers/utils/component-tools';
+import { ISheetClipboardService, PREDEFINED_HOOK_NAME } from '../../services/clipboard/clipboard.service';
+import { ISheetSelectionRenderService } from '../../services/selection/base-selection-render.service';
+import { SheetSkeletonManagerService } from '../../services/sheet-skeleton-manager.service';
+import styles from './index.module.less';
+
+const DEFAULT_PADDING = 2;
+
+const SheetPasteOptions = [
+    { value: 'DEFAULT_PASTE', label: 'rightClick.paste' },
+    { value: 'SPECIAL_PASTE_VALUE', label: 'rightClick.pasteValue' },
+    { value: 'SPECIAL_PASTE_FORMAT', label: 'rightClick.pasteFormat' },
+    { value: 'SPECIAL_PASTE_COL_WIDTH', label: 'rightClick.pasteColWidth' },
+    { value: 'SPECIAL_PASTE_BESIDES_BORDER', label: 'rightClick.pasteBesidesBorder' },
+    { value: 'SPECIAL_PASTE_FORMULA', label: 'formula.operation.pasteFormula' },
+];
+
+const useMenuPosition = (pasteOptionsCache: IPasteOptionCache | null) => {
+    const univerInstanceService = useDependency(IUniverInstanceService);
+    const renderManagerService = useDependency(IRenderManagerService);
+
+    const workbook = useActiveWorkbook();
+
+    const range = pasteOptionsCache?.target.pastedRange;
+    if (!range || !workbook) return null;
+
+    const anchor = {
+        row: range.rows[range.rows.length - 1],
+        col: range.cols[range.cols.length - 1],
+    };
+
+    if (anchor.row < 0 || anchor.col < 0) {
+        return null;
+    }
+
+    const ru = renderManagerService.getRenderById(workbook.getUnitId());
+    const sheetSkeletonManagerService = ru?.with(SheetSkeletonManagerService);
+    const selectionRenderService = ru?.with(ISheetSelectionRenderService);
+
+    const sheetObject = getSheetObject(univerInstanceService, renderManagerService);
+    if (!sheetObject || !selectionRenderService) return null;
+
+    const { scene } = sheetObject;
+    const skeleton = sheetSkeletonManagerService?.getCurrentSkeleton();
```

Comment:
This should reuse positioning logic with autofill. I assume that there is something you can leverage. cc @yuhongz @weird94